### ISSUE TITLE
Email domain is not visible in drop-down list for a long email address 

### ISSUE THREAD

**Issue by vasilicatamas** [mozilla/fxa#3558]
**Note:** 
This issue is a follow-up for 3197

**Build:**
FxA Prod [Train 1.151.2]

**Affected platforms:**
Windows 10
iOS 13.1.2

**Affected versions:**
Firefox 71

**Steps to reproduce:**
1. Launch Firefox a clean profile and click on "Firefox Account" icon from the toolbar.
2. Click on "Turn on Sync..." button.
3. Type a long username followed by the “@” symbol and then type the first character of the email domain.

**Expected results:**
The username is interrupted by ellipses in drop-down list.
Or
A horizontal scroll bar is displayed in drop-down list

**Actual Results:**
The email address is interrupted at the end by ellipses and the email domain is not visible in drop-down list.

**Additional notes:**
This issue is encountered also on iOS 13.1.2

![Screenshot_2](https://user-images.githubusercontent.com/18114210/70062568-f250c200-15ee-11ea-8f2a-f67c787f7213.png)
![unnamed](https://user-images.githubusercontent.com/18114210/70062576-f5e44900-15ee-11ea-9098-ac321044fea6.jpg)

**Comment by LZoog** [comment]
Thanks for testing this out thoroughly!

The only way I can think to address this would be a number of character cutoff - IE, we don't show the `@` suggestions if the number of characters is >= 30. This would be pretty easy to add since we've already got a check to make sure it's at least 2, but we would need to find the number of characters sweetspot. If someone types "m", or another a really wide character, only 20 or 25 times, they'd probably still have this issue.

@davismtl @ryanfeeley we can either close this and say it's behaving as expected since it's native browser functionality, or we can set a character limit for this functionality. Which would you prefer, and if it's the latter, what number would you like to go with?

**Comment by ryanfeeley** [comment]
@LZoog If I'm understanding correctly, you are right that truncating from the end is not ideal as it hides what you are choosing. What would be better is to pick a maximum limit of characters (like 30?) and then truncate in the middle like:

User types:
* thirtyfourcharacters@s

User suggestions include:
* thirtyfo…aracters@someemail.com (the longest one)
* thirtyfor…aracters@bigemail.com
* thirtyfourcharacters@gmail.com (the shortest one)

Wow, this is getting complicated…

**Comment by LZoog** [comment]
I did a little poking around with a [codepen](https://codepen.io/LZoog/pen/jOEEwaO) (I stole most of it from a Stack Overflow answer) and I think it's possible, but level of effort / hackiness might be a little high to get it right honestly, especially considering the differences between all of the browsers.

I'm not sure we can get a super consistent experience for users based on the number of characters - type "th" into the input in that codepen and see the third option. I don't think we can pick a sweetspot number for an ellipsis in the username because if it's not right, it's just going to overflow at the end anyway. Unless we can measure the width of the character string consistently and reliably (probably in some hacky way like in a hidden `div` set to overflow & then we calculate the width vs the width of the textbox, and even then I'm not sure how that'd play across browsers/OSs), it'll be janky. A much easier solution is to disable the functionality at 30+ characters but I'm not sure we want that either.

In other words... Your suggestion might be possible and might also fix #3557 but LoE / potential jank is high for the return considering this is an edgecase. I'm inclined to 1) backlog them both for now, 2) disable functionality at X+ number of characters, or 3) mark this as a "won't fix" as it's native functionality and file an issue with the Android team for #3557 and suggest they replicate iOS functionality pictured here.

**Comment by davismtl** [comment]
This is really an edge case. I'm going to say that this is a `won't fix`. If we put this in the backlog, it will unlikely ever get done.

Feel free to file a bug with the Android team.
